conda/gateways/connection/adapters/jlap/sync_jlap.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""
+Synchronize local patch files with repodata.fly.dev
+"""
+
+import logging
+import os
+from pathlib import Path
+from typing import NamedTuple, cast, Iterator
+
+from requests import Response
+
+from . import truncateable
+from .repodata_proxy import RepodataUrl
+
+log = logging.getLogger(__name__)
+
+
+def line_offsets(path: Path) -> int:
+    """
+    Return byte offset to next-to-last line in path.
+    """
+    sum_val = 0
+    offsets = []
+    with path.open("rb") as data:
+        for line in cast(Iterator, data):
+            offsets.append(sum_val)
+            sum_val += len(line)
+    try:
+        return offsets[-2]
+    except IndexError:
+        return 0
+
+
+class SyncJlap(NamedTuple):
```

Comment:
`SyncJlap` feels like a helper object for the `JlapAdapter`, whats the reasoning for organizing it as a separate object and not defining these helpers directly within `JlapAdapter`?